### PR TITLE
Enhancement/skale 2064 add schain block number check

### PR DESCRIPTION
**Pull request by alexgex**

### CODE REVIEW

File: sla/metrics.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        schains_check = check_schains_for_node(skale, node['id'])
+        metrics['is_offline'] = metrics['is_offline'] | healthcheck | schains_check
+
+    logger.info(f'Received metrics from node ID = {node["id"]}: {metrics}')
+    return metrics
+
+
+def check_schain(skale, schain, node_ip):
+    name = schain['name']
+    addr = "http://" + node_ip + ":" + str(schain['http_rpc_port'])
```
Comment (dmytrotkk) [inline sla/metrics.py]: better to move to the separate function, something like `get_schain_endpoint(ip, rpc_port)`

File: sla/metrics.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    logger.info(f'\nChecking {name}, {addr}')
+
+    try:
+        block_number = skale.web3.eth.blockNumber
+        logger.debug(f"Current block number = {block_number}")
```
Comment (dmytrotkk) [inline sla/metrics.py]: this debug output seems useless - maybe it makes sense to add sChain endpoint or name to it

File: sla/metrics.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    for schain in schains:
+        if check_schain(skale, schain, node_ip) == 1:
+            return 1
+
+    return 0
+
+
+def get_containers_healthcheck(host):
+    """Return 0 if OK or 1 if failed"""
+    url = 'http://' + host + ':' + PORT + HEALTH_REQ_URL
```
Comment (dmytrotkk) [inline sla/metrics.py]: add a separate method for this

File: sla/metrics.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    if ping_parser.parse(
+            result).as_dict()['rtt_avg'] is None or ping_parser.parse(
+                result).as_dict()['packet_loss_count'] > 0:
+        is_offline = True
+        latency = -1
+        logger.info('No connection to host!')
+    else:
+        is_offline = False
+        latency = int((ping_parser.parse(result).as_dict()['rtt_avg']) * 1000)
+        # print('Ping ok!')
```
Comment (dmytrotkk) [inline sla/metrics.py]: remove comment

File: sla/sla_agent.py [hunk 5]
```diff
@@ -76,13 +76,8 @@ def validate_nodes(self, nodes):
             self.logger.info(f'The nodes to be monitored : {nodes}')
 
         for node in nodes:
-            host = GOOD_IP if self.is_test_mode else node['ip']
             if os.system("ping -c 1 " + GOOD_IP + " > /dev/null") == 0:
```
Comment (dmytrotkk) [inline sla/sla_agent.py]: I thought we're using `pingparsing` here, not the system calls

File: sla/metrics.py (hunk 1 above)
Comment (alexgex) [reply sla/metrics.py]: done

File: sla/metrics.py (hunk 2 above)
Comment (alexgex) [reply sla/metrics.py]: I have sChain name output just before this one (with level info). I think there's no sense to duplicate it for debug output but anyway - done :)

File: sla/metrics.py (hunk 3 above)
Comment (alexgex) [reply sla/metrics.py]: done

File: sla/metrics.py (hunk 4 above)
Comment (alexgex) [reply sla/metrics.py]: done

File: sla/sla_agent.py (hunk 5 above)
Comment (alexgex) [reply sla/sla_agent.py]: yep, we're using pingparsing for getting metrics... here was a quick system call just before a node pinging for checking if it's all OK with network (pinging 8.8.8.8 - for example)... anyway - changed for pingparsing too :)

**Review by dmytrotkk — APPROVED** [review]
great!

**Pull request merged**